src/PowerShellEditorServices/Services/Analysis/AnalysisService.cs

```diff
@@ -93,6 +94,8 @@ internal static string GetUniqueIdFromDiagnostic(Diagnostic diagnostic)
 
         private Lazy<PssaCmdletAnalysisEngine> _analysisEngine;
 
+        private FileSystemWatcher _pssaSettingsFileWatcher;
```

Comment:
Because it FileSystemWatcher  inherits from Component, which implements, IDisposable, we should dispose of it in the Dispose method of `AnalysisService`.